embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImpl.kt

```diff
@@ -1,9 +1,60 @@
 package io.embrace.android.embracesdk.capture.envelope.resource
 
+import io.embrace.android.embracesdk.Embrace.AppFramework
 import io.embrace.android.embracesdk.internal.payload.EnvelopeResource
+import io.embrace.android.embracesdk.payload.AppInfo
+import io.embrace.android.embracesdk.payload.DeviceInfo
 
-internal class EnvelopeResourceSourceImpl : EnvelopeResourceSource {
+internal class EnvelopeResourceSourceImpl(
+    private val deviceInfo: DeviceInfo,
+    private val appInfo: AppInfo,
+    private val packageName: String,
+    private val appFramework: AppFramework
+) : EnvelopeResourceSource {
     override fun getEnvelopeResource(): EnvelopeResource {
-        throw NotImplementedError("Not yet implemented")
+        return EnvelopeResource(
+            appVersion = appInfo.appVersion,
+            appEcosystemId = packageName,
+            appFramework = mapFramework(appFramework),
+            buildId = appInfo.buildId,
+            buildType = appInfo.buildType,
+            buildFlavor = appInfo.buildFlavor,
+            environment = appInfo.environment,
+            bundleVersion = appInfo.bundleVersion,
+            sdkVersion = appInfo.sdkVersion,
+            sdkSimpleVersion = appInfo.sdkSimpleVersion!!.toInt(),
```

Comment:
Will this always be set? I'd feel better about using the null coalescing operator here so we don't crash if it is null for whatever reason

Reply:
Is not using appInfo anymore